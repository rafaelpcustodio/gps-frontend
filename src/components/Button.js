import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import StyledButton from './StyledButton'

const StyledText = styled.div`
    color: ${({ selected }) => selected ? 'black' : 'white'};
`

const Button = ( {action, text} ) => {
    return (
        <StyledButton onClick={action}>
            <StyledText>
                {text}
            </StyledText>
        </StyledButton>
    );
}

Button.defaultProps = {
    action: null,
    text: ''
}

Button.propTypes = {
    action: PropTypes.func.isRequired,
    text: PropTypes.string.isRequired
}

export default Button;