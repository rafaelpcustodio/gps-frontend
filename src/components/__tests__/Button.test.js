import configureMockStore from 'redux-mock-store'
import { Map, Record } from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'

import { Provider } from 'react-redux'

import Button from '../Button'

describe('Tests for Button component', () => {
    let store = {}
    let mockStore = {}
    let text = 'text'

    it('Should render without crash', () => {
        mockStore = configureMockStore()
        store = mockStore(Record({
            gpsReducers: Map({
            })
        })())

        const tree = renderer.create(
            <Provider store={ store }>
                <Button
                    action={jest.fn()}
                    text={text}
                />
            </Provider>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })
})
