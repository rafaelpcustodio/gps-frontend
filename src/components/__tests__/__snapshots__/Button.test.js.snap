// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Button component Should render without crash 1`] = `
<button
  className="StyledButton-sc-1z0vb0n-0 ejBCHp"
  onClick={[MockFunction]}
>
  <div
    className="Button__StyledText-ko51rl-0 eowQDx"
  >
    text
  </div>
</button>
`;
